apstools/devices/area_detector_factory.py

```diff
@@ -0,0 +1,330 @@
+"""
+Area Detector Factory
++++++++++++++++++++++
+
+.. autosummary::
+
+   ~ad_creator
+   ~ad_class_factory
+   ~PLUGIN_DEFAULTS
+
+EXAMPLES
+
+Just the camera plugin (uses CamBase, the most basic features)::
+
+    from apstools.devices import ad_creator
+    det = ad_creator("ad:", name="det", class_name="MySimpleAD", ["cam",])
+
+View ADSimDetector image with CA and PVA::
+
+    from ophyd.areadetector import SimDetectorCam
+    from apstools.devices import ad_creator
+
+    det = ad_creator(
+        "ad:", name="det", class_name="MySimDetector",
+        [
+            {"cam": {"class": SimDetectorCam}},
+            "image",
+            "pva",
+        ],
+    )
+
+Record HDF5 images with Eiger detector. Here, both the Eiger detector IOC and
+the Bluesky databroker use the same filesystem mount ``/``::
+
+    from ophyd.areadetector import EigerDetectorCam
+    from apstools.devices import ad_creator
+
+    det = ad_creator(
+        "ad:", name="det", class_name"MyEiger",
+        [
+            {"cam": {"class": EigerDetectorCam}},
+            "image",
+            {"hdf1": {"write_path_template": "/"}},
+        ],
+    )
+
+Override one of the default plugin configurations.  In this case, remove the
+``write_path_template`` and ``read_path_template`` keys from the ``hdf1`` plugin
+support::
+
+    from ophyd.areadetector import EigerDetectorCam
+    from apstools.devices import ad_creator, PLUGIN_DEFAULTS
+
+    plugin_defaults = PLUGIN_DEFAULTS.copy()
+    plugin_defaults["hdf1"].pop("read_path_template", None)
+    plugin_defaults["hdf1"].pop("write_path_template", None)
+
+    det = ad_creator(
+        "ad:", name="det", class_name"MyEiger",
```

Comment:
Make into kw arguments.

Reply:
This is the most problematic of the `plugin_defaults` list structure.  It won't work as written since `plugin_defaults` is a Python `list`.  Which means a unit test should be provided.

Reply:
``PLUGIN_DEFAULTS`` is a dict (https://github.com/BCDA-APS/apstools/blob/61c93bbc81a67b46d61855a57433a6756c20831e/apstools/devices/area_detector_factory.py#L82), and *plugin_defaults* is type-hinted and used as a dict.

Reply:
Right.  Exactly _that_ kind of confusion.

Reply:
Example is correct as written, once the `plugins=` is added.